mod/profile/classes/ElggPlugin/Profile/AnnotationMigration.php

```diff
@@ -21,7 +22,33 @@ public function getVersion() {
 	 * {@inheritdoc}
 	 */
 	public function shouldBeSkipped() {
-		return false;
+		// If there are no user accounts created before the upgrade, e.g. on a new install, skip this
+		$users = elgg_get_entities([
+			'types' => 'user',
+			'wheres' => function(QueryBuilder $qb) {
+				return $qb->compare('e.time_created', '<=', $this->getVersion(), ELGG_VALUE_INTEGER);
```

Comment:
on an upgrade this will skip some users (unwanted) where timecreated is after version but before upgrade date... this will be a problem

Reply:
Real life example?

Reply:
lol, the version of this upgrade is 2017040700 ... every account created after that ts... so almost all accounts of last 12 months

Reply:
besides that it is not even a timestamp

Reply:
Ah, I see. Is there some system time we can use? Do we store any upgrade timestamps?

Reply:
nope.... that's the problem... i have no clue how to fix this just yet... maybe separate this issue from this PR so we can solve this later